hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -118,8 +119,11 @@ public HiveCatalog(
 
     try {
       List<String> tables = clients.run(client -> client.getAllTables(database));
-      List<TableIdentifier> tableIdentifiers = tables.stream()
-          .map(t -> TableIdentifier.of(namespace, t))
+      List<Table> tableObject = clients.run(client -> client.getTableObjectsByName(database, tables));
+      List<TableIdentifier> tableIdentifiers = tableObject.stream()
+          .filter(table -> BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE
+              .equalsIgnoreCase(table.getParameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP)))
```

Comment:
in `org.apache.hadoop.hive.metastore.api.Table` class I do see internally to check for `table.getParameters` is null or not. Do you think that check makes sense here? I'm not completely certain whether `table.getParameters` won't be null

Reply:
Yes,I added the check